spring-integration-core/src/main/java/org/springframework/integration/support/leader/LockRegistryLeaderInitiator.java

```diff
@@ -162,10 +162,33 @@ public LockRegistryLeaderInitiator(LockRegistry locks) {
 	 * @param candidate leadership election candidate
 	 */
 	public LockRegistryLeaderInitiator(LockRegistry locks, Candidate candidate) {
+		this(locks, candidate, Executors.newSingleThreadExecutor(new CustomizableThreadFactory("lock-leadership-")));
+	}
+
+	/**
+	 * Create a new leader initiator. The candidate implementation is provided by the user
+	 * to listen for leadership events and carry out business actions.
+	 * @param locks lock registry
+	 * @param executorService for running leadership election daemon
+	 */
+	public LockRegistryLeaderInitiator(LockRegistry locks, ExecutorService executorService) {
+		this(locks, new DefaultCandidate(), executorService);
+	}
+
+	/**
+	 * Create a new leader initiator. The candidate implementation is provided by the user
+	 * to listen for leadership events and carry out business actions.
+	 * @param locks lock registry
+	 * @param candidate leadership election candidate
+	 * @param executorService for running leadership election daemon
+	 */
+	public LockRegistryLeaderInitiator(LockRegistry locks, Candidate candidate, ExecutorService executorService) {
```

Comment:
I would say this option is not so important.
Let's move it just into a setter!

Reply:
OK, was just trying to keep the changes to a minimum.